crates/interpreter/src/instructions/opcode.rs

```diff
@@ -460,469 +320,336 @@ const IS_PUSH_MASK: u32 = 0x20000000;
-            /* 0xbd */ OpInfo::none(),
-            /* 0xbe */ OpInfo::none(),
-            /* 0xbf */ OpInfo::none(),
-            /* 0xc0 */ OpInfo::none(),
-            /* 0xc1 */ OpInfo::none(),
-            /* 0xc2 */ OpInfo::none(),
-            /* 0xc3 */ OpInfo::none(),
-            /* 0xc4 */ OpInfo::none(),
-            /* 0xc5 */ OpInfo::none(),
-            /* 0xc6 */ OpInfo::none(),
-            /* 0xc7 */ OpInfo::none(),
-            /* 0xc8 */ OpInfo::none(),
-            /* 0xc9 */ OpInfo::none(),
-            /* 0xca */ OpInfo::none(),
-            /* 0xcb */ OpInfo::none(),
-            /* 0xcc */ OpInfo::none(),
-            /* 0xcd */ OpInfo::none(),
-            /* 0xce */ OpInfo::none(),
-            /* 0xcf */ OpInfo::none(),
-            /* 0xd0 */ OpInfo::none(),
-            /* 0xd1 */ OpInfo::none(),
-            /* 0xd2 */ OpInfo::none(),
-            /* 0xd3 */ OpInfo::none(),
-            /* 0xd4 */ OpInfo::none(),
-            /* 0xd5 */ OpInfo::none(),
-            /* 0xd6 */ OpInfo::none(),
-            /* 0xd7 */ OpInfo::none(),
-            /* 0xd8 */ OpInfo::none(),
-            /* 0xd9 */ OpInfo::none(),
-            /* 0xda */ OpInfo::none(),
-            /* 0xdb */ OpInfo::none(),
-            /* 0xdc */ OpInfo::none(),
-            /* 0xdd */ OpInfo::none(),
-            /* 0xde */ OpInfo::none(),
-            /* 0xdf */ OpInfo::none(),
-            /* 0xe0 */ OpInfo::none(),
-            /* 0xe1 */ OpInfo::none(),
-            /* 0xe2 */ OpInfo::none(),
-            /* 0xe3 */ OpInfo::none(),
-            /* 0xe4 */ OpInfo::none(),
-            /* 0xe5 */ OpInfo::none(),
-            /* 0xe6 */ OpInfo::none(),
-            /* 0xe7 */ OpInfo::none(),
-            /* 0xe8 */ OpInfo::none(),
-            /* 0xe9 */ OpInfo::none(),
-            /* 0xea */ OpInfo::none(),
-            /* 0xeb */ OpInfo::none(),
-            /* 0xec */ OpInfo::none(),
-            /* 0xed */ OpInfo::none(),
-            /* 0xee */ OpInfo::none(),
-            /* 0xef */ OpInfo::none(),
-            /* 0xf0  CREATE */ OpInfo::gas_block_end(0),
-            /* 0xf1  CALL */ OpInfo::gas_block_end(0),
-            /* 0xf2  CALLCODE */ OpInfo::gas_block_end(0),
-            /* 0xf3  RETURN */ OpInfo::gas_block_end(0),
-            /* 0xf4  DELEGATECALL */ OpInfo::gas_block_end(0),
-            /* 0xf5  CREATE2 */ OpInfo::gas_block_end(0),
-            /* 0xf6 */ OpInfo::none(),
-            /* 0xf7 */ OpInfo::none(),
-            /* 0xf8 */ OpInfo::none(),
-            /* 0xf9 */ OpInfo::none(),
-            /* 0xfa  STATICCALL */ OpInfo::gas_block_end(0),
-            /* 0xfb */ OpInfo::none(),
-            /* 0xfc */ OpInfo::none(),
-            /* 0xfd  REVERT */ OpInfo::gas_block_end(0),
-            /* 0xfe  INVALID */ OpInfo::gas_block_end(0),
-            /* 0xff  SELFDESTRUCT */ OpInfo::gas_block_end(0),
-        ];
-    };
+const fn opcode_gas_info(opcode: u8, spec: SpecId) -> OpInfo {
```

Comment:
moved gas declarations from macros to const fns.

Reply:
Is this something you'd want to move on the main OPcode macro? i.e make it point to a statement that returns `OpInfo`?

Reply:
I considered doing this but each match arm is pretty substantial for each opcode and would clutter the main invocation